### PR TITLE
Concept for showing boulders on the mapgen preview.

### PR DESCRIPTION
**Pull request by clivepaterson**
So I played around a bit, with the idea of showing the bolder props on the map preview.
I've made it so that each biome can set a colour to use in the preview.

It's took a lot of tries at different things. Having 1 colour to work well on 50% of maps and bad on others..
Having white on white doesn't work well and dark maps need a fairly bright boulder colour, (having an outline helps).

Anyway, I'll grab screen shots of the different biomes, to see what it looks like atm:
Brimstone:
image
image

Desert:
image
image

EarlyAutumn:
image
image

Frithen:
image
image
(this one doesn't look to great actually, now that I'm taking some screen shots)

Mars:
image
image

Moonlight:
image
image
I noticed that moonlight was returning the Mars generator, not sure if it was intended.
So I brought back the moonlight biome, but I adjusted the 'Light.scmlighting' because it's too dark I think.
I can undo this, and set it back to mars, otherwise... this is how it looks
image

Prayer:
image
image

Stones:
image
image

Sunset:
image
image

Syrtis:
image
image

WindingRiver:
image
image

Wonder:
image
image

And last, but not least, Crystalline:
image
image
Which also has some scope creep, to enable the 50 and 150 mass boudlers.
So it looks like this now:
image

Geeeees, so many biomes.....
Anyway's, I found that a different colour per biome gave the most flexibility, however I feel that some are more obvious than others, and some colours at least could use some tweaking...

### CODE REVIEW

File: generator/src/main/java/com/faforever/neroxis/generator/style/StyleGenerator.java [hunk 1]
```diff
@@ -123,12 +123,14 @@ public SCMap generate(GeneratorParameters generatorParameters, long seed) {
...
         CompletableFuture<Void> normalFuture = CompletableFuture.runAsync(textureGenerator::setCompressedDecals);
-        CompletableFuture<Void> previewFuture = CompletableFuture.runAsync(textureGenerator::generatePreview);
+
         CompletableFuture<Void> resourcesFuture = CompletableFuture.runAsync(resourceGenerator::placeResources);
         CompletableFuture<Void> decalsFuture = CompletableFuture.runAsync(decalGenerator::placeDecals);
         CompletableFuture<Void> propsFuture = resourcesFuture.thenAccept(aVoid -> propGenerator.placeProps());
         CompletableFuture<Void> unitsFuture = resourcesFuture.thenAccept(aVoid -> propGenerator.placeUnits());
 
+        CompletableFuture<Void> previewFuture = propsFuture.thenAccept(aVoid -> textureGenerator.generatePreview());
+
```
Comment (clivepaterson) [inline generator/src/main/java/com/faforever/neroxis/generator/style/StyleGenerator.java:133]: So this change prolly deserves explanation.
Due to the new code in 'PreviewGenerator.addReclaimProps()' where it loops over 'map.getProps()'
It basically needs the propsFuture to complete before starting generatePreview()

So I checked the performance impact and from what I can tell it doesn't impact overall performance much. 

This was a worst case map:
neroxis_map_generator_snapshot_rb5zl3islek2s_aiiaeaq
image
Done setHeightMap: 22.56 ms, not found
Done generateResources: 1200.35 ms, not found
Done setCompressedDecals: 795.46 ms, not found
Done generateTextures: 1011.21 ms, not found
Done placeDecals: 1657.39 ms, not found
Done placeProps: 1579.68 ms, not found
Done generatePreview: 163.44 ms, not found
Done setPlacements: 2.55 ms, not found
Pipeline completed!
Map generation done: 5883 ms

This is a map with much less boulders:
neroxis_map_generator_snapshot_px5x6namtp7j2_aiiaeey
image
Done setHeightMap: 24.08 ms, not found
Done generateResources: 782.38 ms, not found
Done setCompressedDecals: 649.94 ms, not found
Done generateTextures: 856.91 ms, not found
Done placeDecals: 1356.56 ms, not found
Done placeProps: 1433.05 ms, not found
Done generatePreview: 144.63 ms, not found
Done setPlacements: 2.82 ms, not found
Pipeline completed!
Map generation done: 5022 ms

**Comment by clivepaterson** [comment]
Ant thoughts @Sheikah45 ?

**Comment by Sheikah45** [comment]
For the aesthetic I mostly just not sure about not just using the game built in colors to indicate reclaim, to also keep the requirements for new biomes small.

**Comment by clivepaterson** [comment]
> For the aesthetic I mostly just not sure about not just using the game built in colors to indicate reclaim, to also keep the requirements for new biomes small.

@Sheikah45 @BlackYps 

Ok, I've removed the `boulderMassColour` from the biomes. It simplifies things a lot.
And I did my best to lookup the build in colour, which is #d18e37 from the lobby from what I can tell.

So I had to darken the grey outline, and I'm aslightly lazy to paste 20 new images.... so I've picked a few which seem good for testing: (To me it seems better)
image
image
image
image
image
image

I feel,
One of the advantages of putting this in the Map Gen preview, is:
- You will be able to see the boulder reclaim information, in the Faf Client's preview, when generating lots of maps.

And the lobby looks like this, (it's different from my Intellij screen shots about, I think is uses different anti-aliasing)
image

**Review by Sheikah45 — APPROVED** [review]
This looks good to me but will let @BlackYps chime in

**Comment by BlackYps** [comment]
Looks good to me as well

**Pull request merged**